### PR TITLE
Hdf5 update

### PR DESCRIPTION
**Pull request by thewtex**
Todos:

- [x] check for new symbols to name mangle on macOS.
- [x] check for new symbols to name mangle on Windows.
- [x] check for clobbering of Emscripten fixes
- [x] remove `_` prefixed mangled symbols

### CODE REVIEW

File: Modules/ThirdParty/HDF5/src/itkhdf5/src/H5make_libsettings.c [hunk 1]
```diff
@@ -43,154 +38,143 @@ static const char *FileHeader = "\n\
...
  *
+ * Return:    void
  *-------------------------------------------------------------------------
  */
 static void
 insert_libhdf5_settings(FILE *flibinfo)
 {
-#if defined(H5_HAVE_EMBEDDED_LIBINFO) && !defined(__EMSCRIPTEN__)
-    FILE *fsettings;	/* for files libhdf5.settings */
+#ifdef H5_HAVE_EMBEDDED_LIBINFO
```
Comment (dzenanz) [inline Modules/ThirdParty/HDF5/src/itkhdf5/src/H5make_libsettings.c]: Is removal of emscripten-related additions intentional? It also appears in a few other places.

File: Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h [hunk 2]
```diff
@@ -3860,7 +3860,1103 @@ e.g. remove mangling for H5Fget_info but leave it for H5Fget_info1 and H5Fget_in
 #define HDstrtoll itk_HDstrtoll
 #define Nflock itk_Nflock
 #define Pflock itk_Pflock
-#define epoch_marker_class itk_epoch_marker_class
+#define _ZN2H510AbstractDsC2Ev itk__ZN2H510AbstractDsC2Ev
```
Comment (dzenanz) [inline Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h]: I think that symbols starting with `_` are belonging to the standard library, and should not be on the mangle list.

Comment (thewtex) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h]: Good to know. I will remove these, then, and add that note to the update instructions.

File: Modules/ThirdParty/HDF5/src/itkhdf5/src/H5make_libsettings.c (hunk 1 above)
Comment (thewtex) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/H5make_libsettings.c]: Good catch. Reviewing / adding...

**Comment by thewtex** [comment]
Verified locally on Emscripten and ready to go :-).

File: Modules/ThirdParty/HDF5/src/itkhdf5/config/cmake/HDF518_Examples.cmake.in [hunk 3]
```diff
@@ -47,7 +47,7 @@ set(ADD_BUILD_OPTIONS "${ADD_BUILD_OPTIONS} -DCTEST_CONFIGURATION_TYPE:STRING=${
 ##################################################################
 
 if(NOT DEFINED INSTALLDIR)
-  set(INSTALLDIR "@CMAKE_INSTALL_PREFIX@")
+  set(INSTALLDIR "C:/Program Files/HDF_Group/@HDF5_PACKAGE_NAME@/@HDF5_PACKAGE_VERSION@")
```
Comment (hjmjohnson) [inline Modules/ThirdParty/HDF5/src/itkhdf5/config/cmake/HDF518_Examples.cmake.in]: I don't think this affect ITK, and I think this is a bug in upstream.

Hard coded absolute paths are bad.

Comment (thewtex) [reply Modules/ThirdParty/HDF5/src/itkhdf5/config/cmake/HDF518_Examples.cmake.in]: :+1:

**Pull request merged**